views-soy/src/main/java/io/micronaut/views/soy/SoyContextMediator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.soy;
+
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * Interface by which Soy render context can be managed and orchestrated by a custom {@link SoyContext} object. Provides
+ * the ability to specify variables for <pre>@param</pre> and <pre>@inject</pre> Soy declarations, and the ability to
+ * override objects like the {@link SoyNamingMapProvider}.
+ *
+ * @author Sam Gammon (sam@momentum.io)
+ * @see SoyContext Default implementation of this interface
+ * @since 1.3.2
+ */
+public interface SoyContextMediator {
```

Comment:
new `SoyContextMediator`, which allows users to specify their own custom logic for supplying render context, injected render context, and/or naming maps